src/Microsoft.Azure.WebJobs.Host/Blobs/Listeners/PollLogsStrategy.cs

```diff
@@ -97,18 +115,76 @@ public void Notify(IStorageBlob blobWritten)
                     break;
                 }
 
-                await NotifyRegistrationsAsync(blob, cancellationToken);
+                notifications.Add(NotifyRegistrationsAsync(blob, failedNotifications, cancellationToken));
             }
+            await Task.WhenAll(notifications);
+
+            // get metrics before the scan start
+            int beforeLogTryEnqueue = _tryEnqueue;
+            int beforeLogSkipped = _skippedEnqueue;
+            int beforeLogFailed = failedNotifications.Count;
+            int beforeLogSuccess = beforeLogTryEnqueue - beforeLogSkipped - beforeLogFailed;
+            long beforeLogTime = _stopWatch.ElapsedMilliseconds;
 
+            List<Task> scanLogs = new List<Task>();
+
+            _tryEnqueue = 0;
+            _skippedEnqueue = 0;
+
+            long[] readTimes = new long[_logListeners.Count];
+            int forLoopCount = 0;
             // Poll the logs (to detect ongoing writes).
             foreach (BlobLogListener logListener in _logListeners.Values)
             {
-                cancellationToken.ThrowIfCancellationRequested();
-
-                foreach (IStorageBlob blob in await logListener.GetRecentBlobWritesAsync(cancellationToken))
+                int index = forLoopCount++;
+                scanLogs.Add(Task.Run(async () =>
                 {
                     cancellationToken.ThrowIfCancellationRequested();
-                    await NotifyRegistrationsAsync(blob, cancellationToken);
+
+                    long before = _stopWatch.ElapsedMilliseconds;
+                    var blobs = await logListener.GetRecentBlobWritesAsync(cancellationToken);
+                    readTimes[index] = _stopWatch.ElapsedMilliseconds - before;
+
+                    List<Task> enqueues = new List<Task>();
+                    foreach (IStorageBlob blob in blobs)
+                    {
+                        cancellationToken.ThrowIfCancellationRequested();
+                        enqueues.Add(NotifyRegistrationsAsync(blob, failedNotifications, cancellationToken));
+                    }
+                    await Task.WhenAll(enqueues);
```

Comment:
Same question here on bounds - what if there were 10,000 new blobs here? Probably don't want to start 10k tasks in parallel. Instead we can do batching.